atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1642,9 +1642,18 @@ private void putMetadataForTablesInternal(
     }
 
     private void putMetadataWithoutChangingSettings(final TableReference tableRef, final byte[] meta) {
+        long ts = System.currentTimeMillis();
+
+        Multimap<Cell, Long> oldVersions = getAllTimestamps(
```

Comment:
*shrug* i don't really care either way. It's not a huge deal, the case you're talking about is most commonly encountered in people without leader election blocks and on Cassandra 1.2 where the lock around this is only per machine, doing drop tables at all the same time, which may occur in the same-cross-machine-millisecond or not. They can also appear ahead of us and because we cherry-picked a situation where this is not well cross-client-locked may or may not find our lower timestamped version to delete, and may accumulate up to n-1 (n=number of concurrent starting up atlas clients) garbage copies in that case.